src/html/mod.rs

```diff
@@ -205,6 +222,19 @@ mod tests {
         assert_eq!(result, vec!["element3", "element2", "element1"]);
     }
 
+    #[test]
+    fn root_matches() {
+        let html = Html::parse_document("<p>element1</p><p>element2</p><p>element3</p>");
+        let selector = Selector::parse("p").unwrap();
+        let result: Vec<_> = html
+            .matches(&selector)
+            .iter()
```

Comment:
The `.iter().rev()` seems extraneous as mainly demonstrates properties of `Vec<_>` which is not implemented in this crate.

Reply:
Updated